src/test/java/com/synopsys/integration/stepworkflow/AbstractConsumingSubStepTest.java

```diff
@@ -0,0 +1,87 @@
+package com.synopsys.integration.stepworkflow;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AbstractConsumingSubStepTest {
+
+    private final String verificationData = "Verification output";
+    private final RuntimeException verificationException = new RuntimeException("Verification runtime exception");
+    private final SubStepResponse<Object> subStepResponse_successOutput = new SubStepResponse<>(true, verificationData, verificationException);
+
+    private static Stream<Arguments> populateSuccessTests() {
+        final String successData = "Success Data";
+        RuntimeException runtimeException = new RuntimeException("Success runtime exception");
+        return Stream.of(
+            Arguments.of(true, "", null),
+            Arguments.of(true, "", runtimeException),
+            Arguments.of(true, successData, null),
+            Arguments.of(true, successData, runtimeException)
+        );
+    }
+
+    private static Stream<Arguments> populateFailureTests() {
+        final String failedData = "Failed Data";
+        RuntimeException runtimeException = new RuntimeException("Failure runtime exception");
+        return Stream.of(
+            Arguments.of(true, null, null),
+            Arguments.of(true, null, runtimeException),
+            Arguments.of(false, null, null),
+            Arguments.of(false, null, runtimeException),
+            Arguments.of(false, "", null),
+            Arguments.of(false, "", runtimeException),
+            Arguments.of(false, failedData, null),
+            Arguments.of(false, failedData, runtimeException)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("populateSuccessTests")
+    public void testAbstractConsumingSubStep_Success(boolean subStepSucceeded, String data, Exception e) {
+        AbstractConsumingSubStep<String> abstractConsumingSubStep = new AbstractConsumingSubStep_TestImpl();
+        SubStepResponse<String> subStepResponse_Input = new SubStepResponse<>(subStepSucceeded, data, e);
```

Comment:
Should `SubStepResponse` be updated to have 2 static methods to construct it, and make the constructor private? One for success and one for failure? It is odd that when things are successful that there is an exception.